File: writer-python/app/api/writer_skills.py
"""列举 app/skills/library 下已加载的 YAML Skill。"""

from __future__ import annotations

from fastapi import APIRouter

from app.skills.loader import SKILL_LIBRARY_DIR, list_skill_summaries

router = APIRouter(tags=["writer"])


@router.get("/api/writer/skills")
def list_writer_skills() -> dict:
    """返回 skills: [{id, label}]；目录为空则为 []。"""
    return {
        "libraryDir": str(SKILL_LIBRARY_DIR),
        "skills": list_skill_summaries(),
    }

File: frontend/src/api/writerSkills.ts
import { apiJson } from "./client";

export type WriterSkillOption = {
  id: string;
  label: string;
};

export type WriterSkillsResponse = {
  libraryDir: string;
  skills: WriterSkillOption[];
};

/** 经 Java 代理 Writer，列出 library 目录下可用 YAML Skill */
export function listWriterSkills(): Promise<WriterSkillsResponse> {
  return apiJson<WriterSkillsResponse>("/api/writer/skills");
}

export type StyleAnalyzeResult = {
  avgSentenceLen: number;
  sampleChars: number;
  styleGuideMd: string;
};

/** 从正文样本生成风格约束 Markdown（经 Java 代理 Writer）。 */
export function postWriterStyleAnalyze(sampleText: string): Promise<StyleAnalyzeResult> {
  return apiJson<StyleAnalyzeResult>("/api/writer/style/analyze", {
    method: "POST",
    body: JSON.stringify({ sampleText }),
  });
}

export type IntentSuggestedAction = {
  action: string;
  detail: string;
};

export type IntentPreviewResult = {
  suggestedActions: IntentSuggestedAction[];
};

/** 自然语言 → 建议下一步操作（经 Java 代理 Writer）。 */
export function postWriterIntentPreview(projectId: string, message: string): Promise<IntentPreviewResult> {
  return apiJson<IntentPreviewResult>("/api/writer/agent/intent-preview", {
    method: "POST",
    body: JSON.stringify({ projectId, message }),
  });
}
